src/Databases/MySQL/MaterializedMySQLSyncThread.cpp

```diff
@@ -165,8 +162,47 @@ static void checkMySQLVariables(const mysqlxx::Pool::Entry & connection, const S
 
         for (size_t index = 0; index < variables_block.rows(); ++index)
         {
-            const auto & error_message_it = variables_error_message.find(variable_name_column->getDataAt(index).toString());
-            const String variable_val = variable_value_column->getDataAt(index).toString();
+            const String variable_name = variable_name_column->getDataAt(index).toString();
+            if variable_name == "version"
+            {
+                version = variable_value_column->getDataAt(index).toString();
+                break;
+            }
+        }
+
+        if (version >= "8.4.0")
+        {
+            variables_error_message = {
+                {"log_bin", "ON"},
+                {"binlog_format", "ROW"},
+                {"binlog_row_image", "FULL"},
+            };
+        }
+        else if (versin >= "8.3.0")
+        {
+            variables_error_message = {
+                {"log_bin", "ON"},
+                {"binlog_format", "ROW"},
+                {"binlog_row_image", "FULL"},
+                {"default_authentication_plugin", "mysql_native_password"},
+            };
+        }
+        else
```

Comment:
Due to peculiarities of string comparison, any version starting from "10.0.0" would end up at this part, not in `>= "8.4.0"`

Reply:
fixed https://github.com/ClickHouse/ClickHouse/pull/63269/commits/834be6de43e8b05a0a7e46639025943f0a4199a9